Assets/MRTK/Core/Utilities/XRSubsystemHelpers.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#if UNITY_2019_3_OR_NEWER
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR;
+#else
+using UnityEngine.Experimental.XR;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// A helper class to provide easier access to active Unity XR SDK subsystems.
+    /// </summary>
+    /// <remarks>These properties are only valid for the XR SDK pipeline.</remarks>
+    public static class XRSubsystemHelpers
+    {
+        private static XRInputSubsystem inputSubsystem = null;
+#if UNITY_2019_3_OR_NEWER
+        private static readonly List<XRInputSubsystem> XRInputSubsystems = new List<XRInputSubsystem>();
+#endif // UNITY_2019_3_OR_NEWER
+
+        /// <summary>
+        /// The XR SDK input subsystem for the currently loaded XR plug-in.
+        /// </summary>
+        public static XRInputSubsystem InputSubsystem
+        {
+            get
+            {
+#if UNITY_2019_3_OR_NEWER
+                if (inputSubsystem == null || !inputSubsystem.running)
+                {
+                    inputSubsystem = null;
+                    SubsystemManager.GetInstances(XRInputSubsystems);
```

Comment:
The docs on these functions are kinda light:

https://docs.unity3d.com/ScriptReference/SubsystemManager.GetInstances.html

I'm guessing all of these will first .Clear() the list before adding things right?

Otherwise this will slowly add more and more things to this list over time.

Reply:
Correct, according to the Unity C# reference, `.Clear()` is called before anything is added: https://github.com/Unity-Technologies/UnityCsReference/blob/61f92bd79ae862c4465d35270f9d1d57befd1761/Modules/Subsystems/SubsystemManager.cs#L26:L29